lang/funcs/cidr.go

```diff
@@ -25,7 +26,7 @@ var CidrHostFunc = function.New(&function.Spec{
 	},
 	Type: function.StaticReturnType(cty.String),
 	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
-		var hostNum int
+		var hostNum *big.Int
```

Comment:
I had honestly completely forgotten that `gocty` knows how to decode into `big.Int`! I'm glad to remember now that it does, cause I was originally expecting this change to be a bit more invasive.

One thing I notice looking back at the test cases for these functions is that they aren't testing the situation where the `newbits`, `hostnum`, `netnum` arguments are fractional numbers. I'm expecting that all of those would be properly rejected as invalid by the `gocty` conversions here (it should say something like "value must be a whole number") but might be worth adding some "expected error" test cases as insurance to future changes that might inadvertently make those panic otherwise.

Reply:
Done and done!